### ISSUE TITLE
Getting "Invalid exponent" (parse_path.dart line 477) on some files

### ISSUE THREAD

**Issue by facundomedica** [dnfield/flutter_svg#32]
First of all, thanks for the good work, this is a very useful package!

I tested with several SVGs exported from Figma (figma.com is like Sketch but runs on the browser) and while most of them worked, some did not.

I've tried using an svg from flaticon.com and works fine, then I export the one in Figma and I get "Invalid exponent". I tried importing the file into Inkscape, use the "Clean up document" thing and still the same.

Example of a faulty svg:

```
<svg width="26" height="26" viewBox="0 0 26 26" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M22.1595 3.80852C19.6789 1.35254 16.3807 -4.80966e-07 12.8727 -4.80966e-07C9.36452 -4.80966e-07 6.06642 1.35254 3.58579 3.80852C1.77297 5.60333 0.53896 7.8599 0.0171889 10.3343C-0.0738999 10.7666 0.206109 11.1901 0.64265 11.2803C1.07908 11.3706 1.50711 11.0934 1.5982 10.661C2.05552 8.49195 3.13775 6.51338 4.72783 4.9391C9.21893 0.492838 16.5262 0.492728 21.0173 4.9391C25.5082 9.38548 25.5082 16.6202 21.0173 21.0667C16.5265 25.5132 9.21893 25.5133 4.72805 21.0669C3.17644 19.5307 2.10538 17.6035 1.63081 15.4937C1.53386 15.0627 1.10252 14.7908 0.66697 14.887C0.231645 14.983 -0.0427272 15.4103 0.0542205 15.8413C0.595668 18.2481 1.81686 20.4461 3.5859 22.1976C6.14623 24.7325 9.50955 26 12.8727 26C16.236 26 19.5991 24.7326 22.1595 22.1976C27.2802 17.1277 27.2802 8.87841 22.1595 3.80852Z" fill="black"/>
<path d="M5.50468 7.22599L5.43239 7.19961V2.75006L5.55078 2.75838C6.48068 2.82416 7.05364 3.10795 7.47194 3.31513C7.71834 3.43728 7.91301 3.5337 8.0838 3.5337C8.62293 3.5337 8.90493 2.827 8.90493 2.44821C8.90493 1.93294 8.29428 1.60853 7.78191 1.42707C7.17789 1.2131 6.33963 1.07148 5.53973 1.04817L5.43239 1.0451V0.372776C5.43239 0.188139 5.16763 1.20242e-07 4.90774 1.20242e-07C4.59678 1.20242e-07 4.40421 0.193502 4.40421 0.372776V1.07925L4.30516 1.08954C3.12532 1.21256 0.381048 1.82798 0.381048 4.88C0.381048 7.49544 2.47941 8.2341 4.3307 8.88585L4.40421 8.91179V14.0339L4.28504 14.0247C2.89693 13.917 2.15407 13.2913 1.61174 12.8345C1.31692 12.586 1.084 12.3899 0.863465 12.3899C0.418744 12.3899 1.21448e-07 13.0014 1.21448e-07 13.4754C1.21448e-07 14.4115 1.67331 15.7978 4.29543 15.8398L4.40421 15.8415V16.599C4.40421 16.7781 4.59678 16.9717 4.90774 16.9717C5.16752 16.9717 5.43239 16.7836 5.43239 16.5991V15.791L5.52768 15.7779C7.91611 15.4495 9.28609 13.9149 9.28609 11.5677C9.2862 8.83201 7.41821 7.92283 5.50468 7.22599ZM4.53134 6.88331L4.38177 6.82738C3.23056 6.39726 2.27755 5.93813 2.27755 4.67052C2.27755 3.61972 3.01389 2.97333 4.40709 2.80095L4.53134 2.78563V6.88331ZM5.43604 13.9812L5.30537 14.0052V9.28413L5.45947 9.34958C6.46211 9.775 7.38991 10.366 7.38991 11.7981C7.38991 12.9749 6.69591 13.7502 5.43604 13.9812Z" transform="translate(8.22925 5.12915)" fill="black"/>
</svg>
```

I know this might be specific to Figma, but I'm following this: `Please submit SVGs this can't render properly (e.g. that don't render here the way they do in chrome), as long as they're not using anything "probably out of scope" (above).`

Thanks!

**Comment by dnfield** [comment]
There is a bug in the exponent validation logic in path_drawing - instead of checking that the exponent is between -37 and 38 (inclusive), it was checking that the exponent was between 37 and 38 (inclusive). I just published a fix in 0.3.4.

At the same time, your example SVG is not encoded as efficiently as it could be - if you run it through svgcleaner, it will fix up your paths a bit so they don't use those weird very small numbers on the C commands:

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="26" height="26" fill="none" viewBox="0 0 26 26">
    <g fill="#000">
        <path d="m22.1595 3.80852c-2.4806-2.45598-5.7788-3.80852048-9.2868-3.80852048-3.50818 0-6.80628 1.35254048-9.28691 3.80852048-1.81282 1.79481-3.04683 4.05138-3.5686011 6.52578-.0910888.4323.1889201.8558.6254611.946.43643.0903.86446-.1869.95555-.6193.45732-2.16905 1.53955-4.14762 3.12963-5.7219 4.4911-4.446262 11.79837-4.446372 16.28947 0 4.4909 4.44638 4.4909 11.6811 0 16.1276-4.4908 4.4465-11.79837 4.4466-16.28925.0002-1.55161-1.5362-2.62267-3.4634-3.09724-5.5732-.09695-.431-.52829-.7029-.96384-.6067-.435325.096-.7096972.5233-.6127495.9543.5414475 2.4068 1.7626395 4.6048 3.5316795 6.3563 2.56033 2.5349 5.92365 3.8024 9.2868 3.8024 3.3633 0 6.7264-1.2674 9.2868-3.8024 5.1207-5.0699 5.1207-13.31919 0-18.38908z"/>
        <path d="m5.50468 7.22599-.07229-.02638v-4.44955l.11839.00832c.9299.06578 1.50286.34957 1.92116.55675.2464.12215.44107.21857.61186.21857.53913 0 .82113-.7067.82113-1.08549 0-.51527-.61065-.83968-1.12302-1.02114-.60402-.21397-1.44228-.35559-2.24218-.3789l-.10734-.00307v-.672324c0-.184637-.26476-.37277588-.52465-.37277588-.31096 0-.50353.19350188-.50353.37277588v.706474l-.09905.01029c-1.17984.12302-3.924112.73844-3.924112 3.79046 0 2.61544 2.098362 3.3541 3.949652 4.00585l.07351.02594v5.12211l-.11917-.0092c-1.38811-.1077-2.13097-.7334-2.6733-1.1902-.29482-.2485-.52774-.4446-.748275-.4446-.444721 0-.86346488.6115-.86346488 1.0855 0 .9361 1.67330988 2.3224 4.29542988 2.3644l.10878.0017v.7575c0 .1791.19257.3727.50353.3727.25978 0 .52465-.1881.52465-.3726v-.8081l.09529-.0131c2.38843-.3284 3.75841-1.863 3.75841-4.2102.00011-2.73569-1.86788-3.64487-3.78141-4.34171zm-.97334-.34268-.14957-.05593c-1.15121-.43012-2.10422-.88925-2.10422-2.15686 0-1.0508.73634-1.69719 2.12954-1.86957l.12425-.01532zm.9047 7.09789-.13067.024v-4.72107l.1541.06545c1.00264.42542 1.93044 1.01642 1.93044 2.44852 0 1.1768-.694 1.9521-1.95387 2.1831z" transform="translate(8.22925 5.12915)"/>
    </g>
</svg>
```

That said, I should have a fix for this in there soon.

**Comment by dnfield** [comment]
Actually by soon I mean now ;)

published in v0.5.1, will commit to github later today (having connectivity issues at the moment).
